### DISCUSSION TITLE
Got the wrong token from ChinesePhonemesTokenizer

### DISCUSSION THREAD

**Discussion by arthurwu4work** [NVIDIA-NeMo/NeMo#7456] [Q&A]
Hi ,

I use the following script to convert input text string to tokens
(according to NeMo/examples/tts/conf/zh/fastpitch_align_22050.yaml)
```
from nemo.collections.tts.g2p.models.zh_cn_pinyin import ChineseG2p
from nemo.collections.common.tokenizers.text_to_speech.tts_tokenizers import ChinesePhonemesTokenizer

input_text = "这些新一代的電腦不只效能惊人。"
g2p = ChineseG2p(phoneme_dict = os.path.join(data_dir, 'zh', "pinyin_dict_nv_22.10.txt"),  word_segmenter = 'jieba')
tokenizer = ChinesePhonemesTokenizer(g2p = g2p, \
                                          punct = True, \
                                          apostrophe = True, \
                                          pad_with_space = True)

token= tokenizer.encode(input_text )
```

However, the code output is different from the NeMo parse function.

**the token I converted**
`[0, 47, 10, 52, 44, 19, 49, 44, 20, 49, 45, 18, 50, 9, 2, 52, 9, 10, 53, 9, 18, 3, 52, 27, 5, 51, 6, 37, 52, 47, 18, 49, 44, 18, 5, 52, 27, 13, 50, 23, 21, 49, 33, 12, 50, 82, 0]`

**the token NeMo parse function converted**   `parsed = spec_generator.parse("这些新一代的電腦不只效能惊人。")`
`[ 0, 37, 34, 44, 30,  8,  4, 41, 30,  7, 13, 41,  7, 42, 19,  2, 44, 19, 34, 45, 19,  8, 33, 13, 44, 13,  3, 43, 16, 25, 44, 37,  7, 41, 30,  8, 3, 44, 13, 31, 29, 42, 22,  7, 29, 41, 40, 31, 13, 42, 74,  0]`

Is there something wrong with my script ?

I use the token I converted to generate the audio.
`[0, 47, 10, 52, 44, 19, 49, 44, 20, 49, 45, 18, 50, 9, 2, 52, 9, 10, 53, 9, 18, 3, 52, 27, 5, 51, 6, 37, 52, 47, 18, 49, 44, 18, 5, 52, 27, 13, 50, 23, 21, 49, 33, 12, 50, 82, 0]`
And the audio file is obviously wrong.
here is the sample audio
[sample.zip](https://github.com/NVIDIA/NeMo/files/12696942/sample.zip)

thank you

**Comment by RobinDong** [comment]
I run below snippet under up-to-date main branch of NeMo and it give the same result:
```python
from nemo.collections.tts.models import FastPitchModel
from nemo.collections.tts.g2p.models.zh_cn_pinyin import ChineseG2p
from nemo.collections.common.tokenizers.text_to_speech.tts_tokenizers import ChinesePhonemesTokenizer

text = "这些新一代的電腦不只效能惊人。"

fastpitch = FastPitchModel.from_pretrained("tts_zh_fastpitch_sfspeech").to("cpu")

parsed = fastpitch.parse(text)
print("A:", list(parsed.numpy()[0]))

g2p = ChineseG2p(phoneme_dict="scripts/tts_dataset_files/zh/24finals/pinyin_dict_nv_22.10.txt", word_segmenter='jieba')
tokenizer = ChinesePhonemesTokenizer(g2p=g2p, punct=True, apostrophe=True, pad_with_space=True)

token = tokenizer.encode(text)
print("B:", token)
```
result:
```
A: [0, 47, 10, 52, 44, 19, 49, 44, 20, 49, 45, 18, 50, 9, 2, 52, 9, 10, 53, 9, 18, 3, 52, 27, 5, 51, 6, 37, 52, 47, 18, 49, 44, 18, 5, 52, 27, 13, 50, 23, 21, 49, 33, 12, 50, 82, 0]
B: [0, 47, 10, 52, 44, 19, 49, 44, 20, 49, 45, 18, 50, 9, 2, 52, 9, 10, 53, 9, 18, 3, 52, 27, 5, 51, 6, 37, 52, 47, 18, 49, 44, 18, 5, 52, 27, 13, 50, 23, 21, 49, 33, 12, 50, 82, 0]
```
